File: examples/e-spring-example/src/main/java/dev/akif/espringexample/crud/Service.java
package dev.akif.espringexample.crud;

import java.util.List;

import e.java.EOr;

public interface Service<DTO, DTOWithId> {
    EOr<List<DTOWithId>> getAll();

    EOr<DTOWithId> getById(long id);

    EOr<DTOWithId> create(DTO dto);

    EOr<DTOWithId> update(long id, DTO dto);

    EOr<DTOWithId> delete(long id);
}

File: examples/e-http4s-example/src/main/scala/dev/akif/ehttp4sexample/common/Service.scala
package dev.akif.ehttp4sexample.common

trait Service[F[_], M, C, U] {
  def getAll: F[List[M]]

  def get(id: Long): F[M]

  def create(create: C): F[M]

  def update(id: Long, update: U): F[M]

  def delete(id: Long): F[M]
}
